src/assets/scripts/client/aircraft/AircraftController.js

```diff
@@ -16,31 +23,110 @@ import { GAME_EVENTS } from '../game/GameController';
 // Temporary const declaration here to attach to the window AND use as internal property
 const aircraft = {};
 
-// TODO: this should be renamed to `AircraftCollection`
 /**
  * @class AircraftController
  */
 export default class AircraftController {
     /**
      * @constructor
+     * @for AircraftController
+     * @param aircraftTypeDefinitionList {array<object>}
+     * @param airlineController {AirlineController}
+     * @param navigationLibrary {NavigationLibrary}
      */
-    constructor() {
+    constructor(aircraftTypeDefinitionList, airlineController, navigationLibrary) {
+        if (!_isArray(aircraftTypeDefinitionList) || _isEmpty(aircraftTypeDefinitionList)) {
+            // eslint-disable-next-line max-len
+            throw new TypeError('Invalid aircraftTypeDefinitionList passed to AircraftCollection. Expected and array but ' +
+                `received ${typeof aircraftTypeDefinitionList}`);
+        }
+
+        // TODO: this may need to use instanceof instead, but that may be overly defensive
+        if (!_isObject(airlineController) || !_isObject(navigationLibrary)) {
```

Comment:
For my own edification, why might that be "overly defensive"? It would seem straightforward to me.